nano/node/node.cpp

```diff
@@ -2727,6 +2727,44 @@ size_t nano::election::last_votes_size ()
 	return last_votes.size ();
 }
 
+void nano::active_transactions::confirm_frontiers (nano::transaction const & transaction_a)
+{
+	// Limit maximum count of elections to start
+	bool representative (node.config.enable_voting && node.wallets.reps_count > 0);
+	/* Check less frequently for non-representative nodes and if previous checks show no unconfirmed frontiers
+	~15 minutes for non-representative nodes, 3.5 minutes for representatives
+	4 times slower if previously all frontiers were confirmed */
+	if (request_confirm_iteration % ((60 - (45 * representative)) * (3 * last_frontier_account.is_zero () + 1)) == 0)
```

Comment:
Let's simplify this logic a bit by calculating a time point when we should check next.
if (std::chrono::steady_clock::now () > = next_frontier_check)

Reply:
Makes sense. Could you check new version?